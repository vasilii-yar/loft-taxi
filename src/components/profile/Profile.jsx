import React from 'react';
import NavBar from "../../components/navbar/NavBar";
import CreditCard from "../../components/profile/CreditCard";
import {Box} from "@material-ui/core";

export default class Profile extends React.Component {
    render() {
        return (
            <>
                <header>
                    <NavBar/>
                </header>
                <main>
                    <section>
                        <Box>
                            <CreditCard/>
                        </Box>
                    </section>
                </main>
            </>
        );
    }
}