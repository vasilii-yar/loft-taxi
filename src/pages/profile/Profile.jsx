import React from 'react';
import NavBar from "../../components/navbar/NavBar";

export default class Profile extends React.Component {
    render() {
        return (
            <>
                <header>
                    <NavBar navigateTo={this.props.navigateTo}/>
                </header>
                <main>
                    <section>

                    </section>
                </main>
            </>
        );
    }
}