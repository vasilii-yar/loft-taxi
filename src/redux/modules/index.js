import authReducer from "./auth/authReducer";
import profileReducer from "./profile/profileReducer";
import {combineReducers} from "redux";


export default combineReducers({
    auth: authReducer,
    profile: profileReducer
});