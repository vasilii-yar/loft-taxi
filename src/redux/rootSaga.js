import {all} from "@redux-saga/core/effects";
import {authSaga} from "./modules/auth/authSaga";
import {profileSaga} from "./modules/profile/profileSaga";
import {addressSaga} from "./modules/address/addressSaga";
import {routeSaga} from "./modules/route/routeSaga";

export default function* rootSaga() {
    yield all([
        authSaga(),
        profileSaga(),
        addressSaga(),
        routeSaga()
    ])
}