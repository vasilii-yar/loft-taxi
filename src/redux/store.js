import {applyMiddleware, createStore} from "redux";
import mainReducer from "./modules"
import {authMiddleware} from "./middleware/authMiddleware";
import {profileMiddleware} from "./middleware/profileMiddleware";

export const store = createStore(mainReducer, applyMiddleware(authMiddleware, profileMiddleware));