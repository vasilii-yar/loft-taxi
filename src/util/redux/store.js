import {applyMiddleware, createStore} from "redux";
import mainReducer from "../../reducers"
import {authMiddleware} from "../../middleware/authMiddleware";

export const store = createStore(mainReducer, applyMiddleware(authMiddleware));