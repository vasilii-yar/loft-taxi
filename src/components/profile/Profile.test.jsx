import React from 'react';
import {render} from "@testing-library/react";
import Profile from "./Profile";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {store} from "../../redux/store";

jest.mock("../navbar/NavBar", () => (
    {
        __esModule: true,
        default: () => <div>Профиль</div>
    }
));
jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
    Map: () => ({})
}));

describe("Profile", () => {
    it("renders correctly", () => {
        const {container} = render(
            <Provider store={store}>
                <Profile/>
            </Provider>
        );
        expect(container.innerHTML).toMatch("Профиль");
    })

    it("renders correctly check by snapshot", () => {
        const tree = renderer.create(
            <Provider store={store}>
                <Profile/>
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
});