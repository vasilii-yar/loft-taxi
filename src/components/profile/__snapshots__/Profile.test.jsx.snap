// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders correctly check by snapshot 1`] = `
Array [
  <header>
    <div>
      Профиль
    </div>
  </header>,
  <main>
    <section>
      <div
        className="MuiBox-root MuiBox-root-1"
      >
        <div
          className="MuiPaper-root credit-card-layout MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            className="MuiBox-root MuiBox-root-2"
            id="PaymentForm"
          >
            <div
              className="rccs"
            >
              <div
                className="rccs__card rccs__card--unknown"
              >
                <div
                  className="rccs__card--front"
                >
                  <div
                    className="rccs__card__background"
                  />
                  <div
                    className="rccs__issuer"
                  />
                  <div
                    className="rccs__cvc__front"
                  >
                    
                  </div>
                  <div
                    className="rccs__number"
                  >
                    •••• •••• •••• ••••
                  </div>
                  <div
                    className="rccs__name"
                  >
                    YOUR NAME HERE
                  </div>
                  <div
                    className="rccs__expiry"
                  >
                    <div
                      className="rccs__expiry__valid"
                    >
                      valid thru
                    </div>
                    <div
                      className="rccs__expiry__value"
                    >
                      ••/••
                    </div>
                  </div>
                  <div
                    className="rccs__chip"
                  />
                </div>
                <div
                  className="rccs__card--back"
                >
                  <div
                    className="rccs__card__background"
                  />
                  <div
                    className="rccs__stripe"
                  />
                  <div
                    className="rccs__signature"
                  />
                  <div
                    className="rccs__cvc"
                  >
                    
                  </div>
                  <div
                    className="rccs__issuer"
                  />
                </div>
              </div>
            </div>
            <form
              className="input-fields"
              onSubmit={[Function]}
            >
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
              >
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <div
                    className="MuiFormControl-root MuiFormControl-fullWidth"
                  >
                    <label
                      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                      data-shrink={false}
                      htmlFor="number"
                    >
                      Номер карты
                    </label>
                    <div
                      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                      onClick={[Function]}
                    >
                      <input
                        aria-invalid={false}
                        className="MuiInputBase-input MuiInput-input"
                        disabled={false}
                        id="number"
                        name="number"
                        onAnimationStart={[Function]}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        required={false}
                        type="number"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <div
                    className="MuiFormControl-root MuiFormControl-fullWidth"
                  >
                    <label
                      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                      data-shrink={false}
                      htmlFor="name"
                    >
                      Имя владельца
                    </label>
                    <div
                      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                      onClick={[Function]}
                    >
                      <input
                        aria-invalid={false}
                        className="MuiInputBase-input MuiInput-input"
                        disabled={false}
                        id="name"
                        name="name"
                        onAnimationStart={[Function]}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        required={false}
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <div
                    className="MuiGrid-root MuiGrid-container"
                  >
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
                    >
                      <div
                        className="MuiFormControl-root MuiFormControl-fullWidth"
                      >
                        <label
                          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                          data-shrink={false}
                          htmlFor="expiry"
                        >
                          Срок действия
                        </label>
                        <div
                          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                          onClick={[Function]}
                        >
                          <input
                            aria-invalid={false}
                            className="MuiInputBase-input MuiInput-input"
                            disabled={false}
                            id="expiry"
                            name="expiry"
                            onAnimationStart={[Function]}
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            required={false}
                            type="string"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
                    />
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
                    >
                      <div
                        className="MuiFormControl-root MuiFormControl-fullWidth"
                      >
                        <label
                          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                          data-shrink={false}
                          htmlFor="cvc"
                        >
                          CVC
                        </label>
                        <div
                          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                          onClick={[Function]}
                        >
                          <input
                            aria-invalid={false}
                            className="MuiInputBase-input MuiInput-input"
                            disabled={false}
                            id="cvc"
                            name="cvc"
                            onAnimationStart={[Function]}
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            required={false}
                            type="number"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <div
                    className="save-button"
                  >
                    <button
                      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                      disabled={false}
                      onBlur={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="submit"
                    >
                      <span
                        className="MuiButton-label"
                      >
                        Сохранить
                      </span>
                      <span
                        className="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </section>
  </main>,
]
`;
